src/language/__tests__/lexer-test.ts

```diff
@@ -168,36 +168,36 @@ describe('Lexer', () => {
   it('errors respect whitespace', () => {
     let caughtError;
     try {
-      lexOne(['', '', '    ?', ''].join('\n'));
+      lexOne(['', '', ' ~', ''].join('\n'));
```

Comment:
please extract changes in this file as separate PR, happy to merge them as is.
This PR is quite big so let's try to extract parts that don't require spec changes.
It will review simpler for everyone from WG.

Also please add test for `?` similar to what we already have for `!`

Reply:
I think I've extracted everything I can. Unfortunately it's not a whole lot, but let me know if I've missed anything. I've also added `?` tests to `lexer-test.ts`.

https://github.com/graphql/graphql-js/pull/3281#issuecomment-975831116

Reply:
Let me know if there are any notes I can write up that might help you and others review this PR. I fully understand that it's quite large!